src/Symfony/Bundle/FrameworkBundle/DependencyInjection/Configuration.php

```diff
@@ -384,13 +384,9 @@ private function addWorkflowSection(ArrayNodeDefinition $rootNode)
                                         ->defaultValue([])
                                         ->prototype('scalar')->end()
                                     ->end()
-                                    ->variableNode('events_to_dispatch')
-                                        ->defaultValue(null)
+                                    ->arrayNode('events_to_dispatch')
```

Comment:
an `arrayNode` must use either `->children` or `prototype` for it to work fine (otherwise, it will be an array node using `children()` but with no allowed children, which is useless)

Reply:
Btw, once you change that to a proper prototype, the `validate` function can then be moved to the prototype instead of looping over the array, which would report the invalid value to a more precise location